tests/scripts/configs_e2e/configs_e2e.go

```diff
@@ -0,0 +1,408 @@
+		return "", fmt.Errorf("error while downloading Bazelisk from %q to %q: %w", bazeliskURL, bazeliskPath, err)
+	}
+
+	return bazeliskPath, nil
+
+}
+
+// runTestBuild runs the remote build using the toolchain configs using Bazelisk to pin the version
+// of Bazel.
+func runTestBuild(ctx context.Context, workingDir, bazelVersion string) error {
+	bazeliskPath, err := downloadBazelisk(workingDir)
+	if err != nil {
+		return fmt.Errorf("failed to download Bazelisk: %w", err)
+	}
+	if err := os.Chmod(bazeliskPath, os.ModePerm); err != nil {
+		return fmt.Errorf("unable to update the permissions of downloaded Bazelisk binary %q to make it executable: %w", bazeliskPath, err)
+	}
+	ctxWithDeadline, cancel := context.WithTimeout(ctx, 3*time.Minute)
+	defer cancel()
+
+	args := []string{
+		// Use a custom output base to ensure Bazel runs with a clean local cache.
+		fmt.Sprintf("--output_base=%s/.bazelcache", workingDir),
+		"build",
+		// This selects all the options specified in the .bazelrc file with config:remote.
+		"--config=remote",
+		// Disable remote caching to ensure the commands constructed from the toolchain configs
+		// are actually valid.
+		"--noremote_accept_cached",
+		"//examples/..."}
+	c := exec.CommandContext(ctxWithDeadline, bazeliskPath, args...)
+	c.Env = append(c.Env, fmt.Sprintf("USE_BAZEL_VERSION=%s", bazelVersion))
+	// Used by Bazelisk to determine where to download Bazel.
+	c.Env = append(c.Env, fmt.Sprintf("XDG_CACHE_HOME=%s/.bazeliskcache", workingDir))
+	c.Dir = workingDir
+	log.Printf("Running '%s %s' with env %v with working directory %q.", bazeliskPath, strings.Join(args, " "), c.Env, workingDir)
+	o, err := c.CombinedOutput()
+	if ctxWithDeadline.Err() == context.DeadlineExceeded {
+		return fmt.Errorf("bazel build timed out")
+	}
+	if err != nil {
+		log.Printf("Output from Bazel:\n%s", string(o))
+		return fmt.Errorf("bazel build failed: %w", err)
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	if len(*manifestURL) == 0 {
+		log.Fatalf("--manifest_url was not specified.")
+	}
+	if len(*configsURL) == 0 {
+		log.Fatalf("--configs_url was not specified.")
+	}
+	if len(*srcRoot) == 0 {
+		log.Fatalf("--src_root was not specified.")
+	}
+	if len(*destRoot) == 0 {
+		log.Fatalf("--dest_root was not specified.")
+	}
+	if len(*rbeInstance) == 0 {
+		log.Fatalf("--rbe_instance was not specified.")
+	}
+	if err := validateRBEInstName(*rbeInstance); err != nil {
+		log.Fatalf("--rbe_instance=%q was invalid: %v", *rbeInstance, err)
+	}
+
+	log.Printf("--manifest_url=%q", *manifestURL)
+	log.Printf("--configs_url=%q", *configsURL)
+	log.Printf("--src_root=%q", *srcRoot)
+	log.Printf("--dest_root=%q", *destRoot)
+	log.Printf("--rbe_instance=%q", *rbeInstance)
+
+	ctx := context.Background()
+
+	m, err := downloadManifest(*manifestURL)
+	if err != nil {
+		log.Fatalf("Unable to download the manifest from %q: %v", *manifestURL, err)
+	}
+	log.Printf("Successfully downloaded the JSON manifest from %s.", *manifestURL)
```

Comment:
```suggestion
	log.Printf("Successfully downloaded the JSON manifest from %q.", *manifestURL)
```

Keeping the trailing . off of URLs helps prevent frustration.  Ditto :403, :405, :407.

Reply:
Done